core/new-gui/src/app/workspace/component/property-editor/property-editor.component.ts

```diff
@@ -112,12 +136,48 @@ export class PropertyEditorComponent {
     // hide the view first and then make everything null
     this.displayForm = false;
 
+    this.showAdvanced = false;
     this.currentOperatorID = undefined;
     this.currentOperatorInitialData = undefined;
     this.currentOperatorSchema = undefined;
-
+    this.tempOperatorSchema = undefined;
   }
 
+  /**
+   * Hides the advanced option fields in the operator by deleting the field.
+   * @param operator
+   */
+  public hideAdvancedFormLayout(operatorSchema: OperatorSchema | undefined): OperatorSchema {
+    if (! operatorSchema) {
+      throw new Error('Parameter operator schema is undefined');
+    }
+    this.tempOperatorSchema = cloneDeep(this.currentOperatorSchema);
+    if (! this.currentOperatorSchema) {
+      throw new Error('Current operator schema is undefined');
```

Comment:
move this check before the line `this.tempOperatorSchema = cloneDeep(this.currentOperatorSchema);` to capture the error before assigning new values to `this.tempOperatorSchema`.